Wikipedia/Code/SearchViewController.swift

```diff
@@ -123,8 +123,11 @@ class SearchViewController: ArticleCollectionViewController, UISearchBarDelegate
                 guard searchTerm == self.searchBar.text else {
                     return
                 }
-                self.resultsViewController.wmf_showEmptyView(of: WMFEmptyViewType.noSearchResults, action: nil, theme: self.theme, frame: self.resultsViewController.view.bounds)
-                WMFAlertManager.sharedInstance.showErrorAlert(error as NSError, sticky: false, dismissPreviousAlerts: true, tapCallBack: nil)
+                
+                let emptyViewType: WMFEmptyViewType = (error as NSError).wmf_isNetworkConnectionError() ? .noInternetConnection : .noSearchResults
+
+                self.resultsViewController.wmf_showEmptyView(of: emptyViewType, action: nil, theme: self.theme, frame: self.resultsViewController.view.bounds)
```

Comment:
Is this missing a corresponding `wmf_hideEmptyView` somewhere? Probably on search success?

Reply:
Seemed be resetting already... guessing it did so to ensure the "No results found" empty view was reset on search success. Will double-check.

Reply:
Also, `self.resultsViewController.view.bounds` is likely the cause of the sizing issue. You could hook into `emptyViewType` and `updateEmptyState()` provided by the superclass `ColumnarCollectionViewController` and the frame should be updated correctly.

Reply:
oh cool will check...

Reply:
Doesn't work because of a `guard` in `updateEmptyState` so it never gets to `isEmptyDidChange`... but even if I call `isEmptyDidChange` directly the frame is wrong... looking at why now...

Reply:
Fixed empty state update to handle changes from one empty type to another and put calls to update after the results view controller reloads data.